data-prepper-plugins/otel-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSourceTest.java

```diff
@@ -411,6 +412,113 @@ void testHttpFullJsonWithCustomPathAndAuthHeader_with_unsuccessful_response() th
+        when(oTelMetricsSourceConfig.getPath()).thenReturn(TEST_PATH);
+    
+        configureObjectUnderTest();
+        SOURCE.start(buffer);
+    
+        final String invalidUsername = "wrong_user";
+        final String invalidPassword = "wrong_password";
+        final String invalidCredentials = Base64.getEncoder()
+                .encodeToString(String.format("%s:%s", invalidUsername, invalidPassword).getBytes(StandardCharsets.UTF_8));
+    
+        final String transformedPath = "/" + TEST_PIPELINE_NAME + "/v1/metrics";
+    
+        WebClient.of().prepare()
+                .post("http://127.0.0.1:21891" + transformedPath)
+                .content(MediaType.JSON_UTF_8, JsonFormat.printer().print(createExportMetricsRequest()).getBytes())
+                .header("Authorization", "Basic " + invalidCredentials)
+                .execute()
+                .aggregate()
+                .whenComplete((response, throwable) -> assertSecureResponseWithStatusCode(response, HttpStatus.UNAUTHORIZED, throwable))
+                .join();
+    }
+    
+    @Test
+    void testGrpcRequestWithInvalidCredentials_with_unsuccessful_response() throws Exception {
+        when(httpBasicAuthenticationConfig.getUsername()).thenReturn(USERNAME);
+        when(httpBasicAuthenticationConfig.getPassword()).thenReturn(PASSWORD);
+        final GrpcAuthenticationProvider grpcAuthenticationProvider = new GrpcBasicAuthenticationProvider(httpBasicAuthenticationConfig);
+
+        when(pluginFactory.loadPlugin(eq(GrpcAuthenticationProvider.class), any(PluginSetting.class)))
+                .thenReturn(grpcAuthenticationProvider);
+        when(oTelMetricsSourceConfig.getAuthentication()).thenReturn(new PluginModel("http_basic",
+                Map.of(
+                        "username", USERNAME,
+                        "password", PASSWORD
+                )));
+        configureObjectUnderTest();
+        SOURCE.start(buffer);
+
+        final String invalidUsername = "wrong_user";
+        final String invalidPassword = "wrong_password";
+        final String invalidCredentials = Base64.getEncoder()
+                .encodeToString(String.format("%s:%s", invalidUsername, invalidPassword).getBytes(StandardCharsets.UTF_8));
+
+        final MetricsServiceGrpc.MetricsServiceBlockingStub client = Clients.builder(GRPC_ENDPOINT)
+                .addHeader("Authorization", "Basic " + invalidCredentials)
+                .build(MetricsServiceGrpc.MetricsServiceBlockingStub.class);
+
+        final StatusRuntimeException actualException = assertThrows(StatusRuntimeException.class, () -> client.export(createExportMetricsRequest()));
+
+        assertThat(actualException.getStatus(), notNullValue());
+        assertThat(actualException.getStatus().getCode(), equalTo(Status.Code.UNAUTHENTICATED));
+    }
+
+    @Test
+    void testHttpsFullJsonWithCustomPathAndAuthHeaderAndSsl_with_unsuccessful_response() throws InvalidProtocolBufferException {
```

Comment:
Are we testing missing credentials (username/password) in this test case? If yes, I would suggest rename the test. If not, missing credentials in client can have a confounding factor causing unauth.

Reply:
this is meant for testing that if SSL is enabled on the server and a client attempts to connect without using SSL/TLS ], the connection will fail as expected. Made changes accordingly